app/classes/importers/photosynthesis_measurement_importer.rb

```diff
@@ -0,0 +1,65 @@
+class PhotosynthesisMeasurementImporter < RowImporter
+
+  def object
+    @pm
+  end
+
+  def read_row(values, logger)
+    @pm = find_or_new({
+      #########################################################################
+      #
+      # The code reader is going to need to do a      CSP -> tree code lookup
+      # All the data we will need to make I9999999 trees work is also in that
+      # table
+      #
+      #########################################################################
+      :leaf_part => find_or_create_leaf_part(values[1]),
+      :pm_type => values[3]
+    })
+    attempt_to_overwrite!(@pm)
+
+    @pm.filename = values[0]
+    @pm.code = values[1]
+    @pm.area_corr = values[2]
+    begin
+      @pm.date = Date.strptime(values[4], "%d/%m/%y")
+    rescue ArgumentError
+      @pm.date = nil
```

Comment:
I can't remember why I'm rescuing a dirty date, also it should never be blank so I'll just strip out the rescue.
Now, for time, likeAlexander Shenkin said in his E. mail "Photosynthesis measurements should have a time associated with them, but for some reason, some of the seem to be NA (though there is a date). Can we put NA instead of midnight? Will the column accept that?"
I intended this to let "NA" or "na" or "none" to slip through and show up in the database as NULL... bearing in mind this is a datetime field so "NA" is not possible. I'll change this so that it only lets an explicit "NA" through.